intelmq/lib/bot_debugger.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities for debugging intelmq bots.
+
+BotDebugger is called via intelmqctl. It starts a live running bot instance,
+leverages logging to DEBUG level and permits even a non-skilled programmer
+who may find themselves puzzled with Python nuances and server deployment twists
+to see what's happening in the bot and where's the error.
+
+Depending on the subcommand received, the class either
+ * starts the bot as is (default)
+ * processes single message, either injected or from default pipeline (process subcommand)
+ * reads the message from input pipeline or send a message to output pipeline (message subcommand)
+"""
+import logging
+import json
+
+from intelmq.lib.message import Event
+from importlib import import_module
+from intelmq.lib.utils import StreamHandler
+from intelmq.lib.message import Event
+
+
+class BotDebugger:
+        def leverageLogger(self, level = logging.DEBUG):
+            self.instance.logger.setLevel(level)
+            for h in self.instance.logger.handlers:
+                if isinstance(h, StreamHandler):
+                    h.setLevel(level)
+
+        def __init__(self, module_path, bot_id, run_subcommand = None, message_kind = None, dryrun = None, msg = None):
+            module = import_module(module_path)
+            bot = getattr(module, 'BOT')
+            self.instance = bot(bot_id)
```

Comment:
Is that really so bad? Why? Is that a bottleneck, or you mind the initialisation messages in the console? ("deduplicator-expert-cz: DeduplicatorExpertBot initialized with id deduplicator-expert and version 3.5.2 (default, Nov 17 2016, 17:05:23) as process 5560.") I may just suppress the messages.
There is so much things I have to implement if I want to connect to bot's pipelines without acually calling bot.init. So much of code that had to be reused; and I may do some small errors that would make the connection differ for debug session and for normal lifecycle of a bot. Loading configuration from different files, manually calling PipelineFactory...